internal/registry/registry.go

```diff
@@ -20,64 +21,72 @@ import (
 
 func SaveImage(imagePath string, image kubernetes.ContainerImage) error {
 	imageMap := map[string]v1.Image{}
-
 	o := crane.GetOptions()
 
-	if len(image.Auth) > 0 {
-		cfg, err := ResolveAuthConfig(image)
-		empty := types.AuthConfig{}
+	for pullSecretIndex, pullSecret := range image.PullSecrets {
 
-		if err != nil {
-			return err
-		}
+		if len(pullSecret.SecretCredsData) > 0 {
+			cfg, err := ResolveAuthConfigWithPullSecretIndex(image, pullSecretIndex)
```

Comment:
Please directly use the `pullSecret` here instead of accessing it with the index.

Reply:
made some changes to the pullSecret iteration handling